tidy3d/components/data/dataset.py

```diff
@@ -985,7 +1174,404 @@ def interp(
         # VTK arrays are the z-y-x order, reorder interpolation results to x-y-z order
         values_reordered = np.transpose(np.reshape(values_numpy, shape[::-1]), (2, 1, 0))
 
-        return SpatialDataArray(values_reordered, coords=dict(x=x, y=y, z=z), name=self.values.name)
+        return values_reordered
+
+    def _interp_py(
+        self,
+        x: ArrayLike,
+        y: ArrayLike,
+        z: ArrayLike,
+        fill_value: float,
+        max_samples_per_step: int = DEFAULT_MAX_SAMPLES_PER_STEP,
+        max_cells_per_step: int = DEFAULT_MAX_CELLS_PER_STEP,
+        rel_tol: float = 1e-10,
+        axis_ignore: Union[Axis, None] = None,
+    ) -> ArrayLike:
+        """Interpolate data at provided x, y, and z using vectorized python implementation.
+
+        Parameters
+        ----------
+        x : Union[float, ArrayLike]
+            x-coordinates of sampling points.
+        y : Union[float, ArrayLike]
+            y-coordinates of sampling points.
+        z : Union[float, ArrayLike]
+            z-coordinates of sampling points.
+        fill_value : float
+            Value to use when filling points without interpolated values.
+        max_samples_per_step : int = 1e4
+            Max number of points to interpolate at per iteration.
+        max_cells_per_step : int = 1e4
+            Max number of cells to interpolate from per iteration.
+        rel_tol : float = 1e-10
+            Relative tolerance when comparing coordinates.
+        axis_ignore : Union[Axis, None] = None
+            For interpolati
```

Comment:
comment got cutoff. I guess axis_ignore is None for Tetrahedral and normal axis for Triangular?

Reply:
that's exactly right, fixed.

Reply:
also, to make code a bit more clear, I renamed the main internal interpolation function into `_interp_py_general` to highlight it is implemented for both 2D and 3D cells simulateneously. Then `TriangularGridDataset` and `TetrahedralGridDataset` each have their own `_interp_py` that wraps `_interp_py_general`.